code/modules/power/engines/supermatter/supermatter.dm

```diff
@@ -803,6 +808,17 @@
 				else
 					to_chat(user, "<span class='warning'>You fail to extract a sliver from [src]! [I] isn't sharp enough anymore.</span>")
 		return
+
+	if(istype(I, /obj/item/twohanded/supermatter))
+		if(ishuman(user))
+			var/mob/living/carbon/human/M = user
+			to_chat(M, "<span class='notice'>You begin to carve off a sliver of [src] with [I]...</span>")
+			if(I.use_tool(src, M, 10 SECONDS, volume = 100))
+				to_chat(M, "<span class='danger'>[src] seethes with energy as you finish cutting off a sliver!</span>")
+				matter_power += 600
+				new /obj/item/nuke_core/supermatter_sliver(drop_location())
```

Comment:
lots of copy pasted code from the if above. would rather you create a generic `carve_sliver` proc on `/obj/machinery/atmospherics/supermatter_crystal` and try and reduce the amount of copy paste here.

Reply:
The code is similar, but the scalpel has a lot more logic here (checking if the scalpel is still sharp, seeing if you have SM tongs, etc) than the halberd - it'd look about the same copy-paste wise, just in a different location, I believe

Reply:
just contain that logic outside of the general proc then. that entire chunk of code from line 813 to 819 can be generalised